src/c++/perf_analyzer/genai-pa/genai_pa/llm_metrics.py

```diff
@@ -121,6 +121,65 @@ def __repr__(self):
         attr_strs = ",".join([f"{k}={v}" for k, v in self.__dict__.items()])
         return f"Statistics({attr_strs})"
 
+    def _is_seconds_field(self, field: str):
+        seconds_fields = [
+            "inter_token_latency",
+            "time_to_first_token",
+            "end_to_end_latency",
+        ]
+        return field in seconds_fields
+
+    def _use_exact_length(self, value: str):
+        exact_value_length = 17
+        formatted_value = f"{value:.{exact_value_length}f}"
+        if len(formatted_value) < exact_value_length:
+            return formatted_value.ljust(exact_value_length)
+        return formatted_value[:exact_value_length]
+
+    def pretty_print(self):
+        field_stats = {}
+        print("Output:")
+
+        for key, value in self.__dict__.items():
+            if key.startswith(
+                (
+                    "p25_",
+                    "p50_",
+                    "p75_",
+                    "p90_",
+                    "p95_",
+                    "p99_",
+                    "avg_",
+                    "min_",
+                    "max_",
+                    "std_",
+                )
+            ):
+                stat, field = key.split("_", 1)
+                stat = stat.replace("_", " ")
+
+                if field not in field_stats:
+                    field_stats[field] = {}
+                field_stats[field][stat] = value
+
+        for field, stats in field_stats.items():
+            is_seconds_field = self._is_seconds_field(field)
```

Comment:
i worry that saying seconds here inadvertently states units when that may not be the case.
maybe _is_time_field or _is_duration_field?
I am not sure

Reply:
Good call. Updated to time field.